packages/cdktf/lib/terraform-variable.ts

```diff
@@ -0,0 +1,170 @@
+import { Construct, Node } from "constructs";
+import { TerraformElement } from "./terraform-element";
+import { keysToSnakeCase, deepMerge } from "./util"
+import { Token } from "./tokens";
+
+export interface IVariableType {
+    /**
+     * @internal
+     */
+    _render(): string;
+}
+
+export class PrimitiveVariableType implements IVariableType {
+    public static readonly STRING = new PrimitiveVariableType('string');
+    public static readonly NUMBER = new PrimitiveVariableType('number');
+    public static readonly BOOL = new PrimitiveVariableType('bool');
+    public static readonly ANY = new PrimitiveVariableType('any');
```

Comment:
Could this be an alternative way to express this? 

```ts
type PrimitiveVariableType = "string" | "number" | "bool" | "any"
```

Reply:
I can double check, but I don't believe jsii supports that. 

It also makes the synth a little more complicated since it needs a special case (I had a similar version with an enum).

Reply:
See other comment https://github.com/hashicorp/terraform-cdk/pull/277#issuecomment-685965999